test/integration/client-side-encryption/client_side_encryption.prose.test.js

```diff
@@ -1424,4 +1424,245 @@ describe('Client Side Encryption Prose Tests', metadata, function () {
+        try {
+          result = await collection.aggregate([]).toArray();
+        } catch (error) {
+          expect(error.code).to.equal(123);
+        }
+        expect(result).to.not.exist;
+        expect(aggregateFailed.failure.code).to.equal(123);
+        expect(aggregateStarted).to.exist;
+      });
+    });
+
+    context('Case 2: Network Error', function () {
+      beforeEach(async function () {
+        // Use ``setupClient`` to configure the following failpoint:
+        //    {
+        //         "configureFailPoint": "failCommand",
+        //         "mode": {
+        //             "times": 1
+        //         },
+        //         "data": {
+        //             "errorCode": 123,
+        //             "closeConnection": true,
+        //             "failCommands": [
+        //                 "aggregate"
+        //             ]
+        //         }
+        //     }
+        await setupClient
+          .db()
+          .admin()
+          .command({
+            configureFailPoint: 'failCommand',
+            mode: {
+              times: 1
+            },
+            data: {
+              errorCode: 123,
+              closeConnection: true,
+              failCommands: ['aggregate']
+            }
+          });
+      });
+
+      it('expects an error and a command failed event', async function () {
+        // Use ``encryptedClient`` to run an aggregate on ``db.decryption_events``.
+        // Expect an exception to be thrown from the network error. Expect a CommandFailedEvent.
+        const collection = encryptedClient.db('').collection('decryption_events');
+        let result;
+        try {
+          result = await collection.aggregate([]).toArray();
+        } catch (error) {
+          expect(error.message).to.include('closed');
+        }
+        expect(result).to.not.exist;
+        expect(aggregateFailed.failure.message).to.include('closed');
+        expect(aggregateStarted).to.exist;
+      });
+    });
+
+    context('Case 3: Decrypt Error', function () {
+      it('errors on decryption but command succeeds', async function () {
+        // Use ``encryptedClient`` to insert the document ``{ "encrypted": <malformedCiphertext> }``
+        // into ``db.decryption_events``.
+        // Use ``encryptedClient`` to run an aggregate on ``db.decryption_events``.
+        // Expect an exception to be thrown from the decryption error.
+        // Expect a CommandSucceededEvent. Expect the CommandSucceededEvent.reply
+        // to contain BSON binary for the field
+        // ``cursor.firstBatch.encrypted``.
+        const collection = encryptedClient.db('').collection('decryption_events');
+        await collection.drop();
+        await collection.insertOne({ encrypted: malformedCiphertext });
+        let result;
+        try {
+          result = await collection.aggregate([]).toArray();
+        } catch (error) {
+          expect(error.message).to.include('HMAC validation failure');
```

Comment:
This is basically showing that we generate command monitoring events before we decrypt as we error during decryption but still get a command succeeded event. Kind of funny but the most secure way to do it I guess.